conda_build/metadata.py

```diff
@@ -282,6 +282,32 @@ def get_contents(meta_path):
     return contents
 
 
+def handle_config_version(ms, ver):
+    """
+    'ms' is an instance of MatchSpec, and 'ver' is the version from the
+    configuration, e.g. for ms.name == 'python', ver = 26 or None,
+    return a (sometimes new) MatchSpec object
+    """
+    if ms.strictness == 3:
+        return ms
+
+    if ms.strictness == 2:
+        if ms.spec.split()[1] == 'x.x':
+            if ver is None:
+                raise RuntimeError("'%s' requires external setting" % ms.spec)
+            # (no return here - proceeds below)
+        else: # regular version
+            return ms
+
+    if ver is None or (ms.strictness == 1 and ms.name == 'numpy'):
```

Comment:
Well if we don't special case numpy, i.e. make requiring the `x.x` when the version is important, then every time you have a python dependency, you have to say `python x.x`, which would break almost all existing recipes.